codeland.html

```diff
@@ -572,42 +572,57 @@
               .then(sleep(time)
               .then(() => poll(promiseFn, time)));
 
-const fetchNowPlayingData = url => (
-  fetch(url)
-    .then((resp) => resp.json())
-    .then(function(data) {
-        updateTalkInfo(data.now_playing.talk_id);
-    })
-    .catch(function(error) {
-        console.log(error);
-    })
+// Update the currently playing section based on schedule and current time
+const updateNowPlaying = () => (
+    new Promise(resolve => updateTalkInfo())
+      .catch(function(error) {
+          console.log(error);
+      })
 );
 
+// Look for updated schedule data
 const fetchScheduleData = url => (
   fetch(url)
     .then((resp) => resp.json())
     .then(function(data) {
-        scheduleData = data.schedule;
+        scheduleData = calculateTimes(data.schedule);
         updateScheduleInfo(scheduleData);
     })
```

Comment:
Idea/Suggestion:

Adding `.then(updateNowPlaying)` to the promise chain inside `fetchScheduleData` function so it would look like:

```
fetch(url)
    .then((resp) => resp.json())
    .then(function(data) {
        scheduleData = calculateTimes(data.schedule);
        updateScheduleInfo(scheduleData);
    })
    .then(updateNowPlaying)
    .catch(function(error) {
        console.log(error);
    })
```

Because whenever we fetch the schedule and the HTML is recreated we lose the "currently playing" until the 5s refresh is triggered. This isn't the end of the world but it may look like a "flash" for a couple seconds (where there's no active talk and then it's highlighted when `updateNowPlaying` is triggered).

Reply:
Oh this is a very good point and excellent solution, thanks!

(Thinking in promises makes my head hurt!)